tests/onegov/org/test_views_search.py

```diff
@@ -147,13 +188,82 @@ def test_search_recently_published_object(client_with_es):
     transaction.commit()
 
     client.get(url)
-
     sleep(5)
 
-    assert 'fulltext' in client.get('/search?q=wow')
-    assert 'fulltext' not in anom.get('/search?q=wow')
-    assert 'It is pretty awesome' in client.get('/search?q=fulltext')
-    assert 'It is pretty awesome' not in anom.get('/search?q=fulltext')
+    page = session.query(Page).filter(
+        Page.title == "Now supporting fulltext search"
+    ).one()
+    assert page.access == 'public'
+    assert page.published == False
+    assert page.es_public == False
+
+    # elasticsearch
+    assert 'fulltext' in client.get('/search-postgres?q=wow')
+    assert 'fulltext' not in anom.get('/search-postgres?q=wow')
+    assert 'is pretty awesome' in client.get('/search-postgres?q=fulltext')
+    assert 'is pretty awesome' not in anom.get('/search-postgres?q=fulltext')
+
+    # postgres
+    assert 'fulltext' in client.get('/search-postgres?q=wow')
+    assert 'fulltext' not in anom.get('/search-postgres?q=wow')
+    assert 'is pretty awesome' in client.get('/search-postgres?q=fulltext')
+    assert 'is pretty awesome' not in anom.get('/search-postgres?q=fulltext')
+
+
+def test_search_recently_published_object_freeze(client_with_es):
```

Comment:
I don't understand why test_search_recently_published_object and tests/onegov/org/test_views_search.py::test_search_recently_published_object_freeze are failing. I introduced the later just to test the same in a different way but it fails as well.
Doing the steps on the UI, the News are only listed when they should, as long as it is within the publication window or after publication start date/time (in case of no end date).
My assumption is that the sql expressions for es_public are not a fully correct causing this issue but couldn't figure it out.
Don't forget to use search-postgres for search queries to use postgres

Reply:
I believe there are some cronjobs that have to run in order for recently published or unpublished models to behave correctly, although I can't recall if that is something elastic search specific or if postgres fts would also be affected by this cronjob.

So only freezing the time might not be enough, you may also need to invoke the cronjob. However, it is very likely that some of the hybrid properties are incorrect. But it will be difficult to test them all and find the culprit(s). It might be more fruitful to try a different approach, as described in my other comment.

Reply:
The cronjob only triggers a re-indexing of the object (meant for ES) but does not impact the postgres data. So no, this is not helping...